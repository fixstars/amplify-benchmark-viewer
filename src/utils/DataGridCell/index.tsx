import { LinkButton } from 'components/atoms/LinkButton'

export interface ProblemInstances {
  readonly [key: string]: ReadonlyArray<string> | undefined
}

const renderClientsCell = (key: string, clients: ReadonlyArray<string>) => {
  const getBackgroundColor = (client: string): string => {
    let backgroundColor = '#FF66FF'

    switch (client) {
      case 'FixstarsClient':
        backgroundColor = '#0F66DC'
        break
      case 'ToshibaClient':
        backgroundColor = '#f57bae'
        break
      case 'ToshibaSQBM2Client':
        backgroundColor = '#e75e75'
        break
      case 'GurobiClient':
        backgroundColor = '#e5f97c'
        break
      case 'DWaveSamplerClient':
        backgroundColor = '#1b000d'
        break
      case 'LeapHybridSamplerClient':
        backgroundColor = '#f6a695'
        break
      case 'FujitsuDASolverClient':
        backgroundColor = '#692981'
        break
      case 'FujitsuDASolverExpertClient':
        backgroundColor = '#3cde29'
        break
      case 'FujitsuDAPTSolverClient':
        backgroundColor = '#591414'
        break
      case 'FujitsuDAMixedModeSolverClient':
        backgroundColor = '#d66594'
        break
      case 'FujitsuDA2SolverClient':
        backgroundColor = '#2d884a'
        break
      case 'FujitsuDA2SolverExpertClient':
        backgroundColor = '#400763'
        break
      case 'FujitsuDA2PTSolverClient':
        backgroundColor = '#9d19f4'
        break
      case 'FujitsuDA2MixedModeSolverClient':
        backgroundColor = '#fcdeb9'
        break
      case 'FujitsuDA3SolverClient':
        backgroundColor = '#718de6'
        break
      case 'FujitsuDA4SolverClient':
        backgroundColor = '#1f4c48'
        break
      case 'HitachiClient':
        backgroundColor = '#909743'
        break
      case 'ABSClient':
        backgroundColor = '#2dc1c8'
        break
      case 'QiskitClient':
        backgroundColor = '#c17913'
        break
      case 'QulacsClient':
        backgroundColor = '#b2a1f6'
        break
    }

    return backgroundColor
  }

  return clients.map((client) => (
    <div key={`${key}_${client}`} style={{ marginRight: 10 }}>
      <LinkButton
        label={client}
        link={`/clients/${client}`}
        backgroundColor={getBackgroundColor(client)}
      />
    </div>
  ))
}

const renderProblemInstancesCell = (
  key: string,
  problemInstances: ProblemInstances,
) => {
  const getBackgroundColor = (client: string): string => {
    let backgroundColor = '#FF66FF'

    switch (client) {
      case 'pr124':
        backgroundColor = '#0F66DC'
        break
      case 'pr136':
        backgroundColor = '#f57bae'
        break
      case 'pr144':
        backgroundColor = '#e75e75'
        break
      case 'si175':
        backgroundColor = '#e5f97c'
        break
      case 'pr299':
        backgroundColor = '#1b000d'
        break
      case 'sko56':
        backgroundColor = '#f6a695'
        break
      case 'sko100a':
        backgroundColor = '#692981'
        break
      case 'tai150b':
        backgroundColor = '#3cde29'
        break
      case 'G65':
        backgroundColor = '#591414'
        break
      case 'G77':
        backgroundColor = '#d66594'
        break
      case 'G81':
        backgroundColor = '#2d884a'
        break
      case '16x16_1_000':
        backgroundColor = '#400763'
        break
      case '25x25_000':
        backgroundColor = '#9d19f4'
        break
      case 'B-n31-k5':
        backgroundColor = '#fcdeb9'
        break
      case 'F-n45-k4':
        backgroundColor = '#718de6'
        break
      case 'burma14':
        backgroundColor = '#1f4c48'
        break
      case 'ulysses16':
        backgroundColor = '#909743'
        break
      case 'bayg29':
        backgroundColor = '#2dc1c8'
        break
      case 'kroA100':
        backgroundColor = '#c17913'
        break
      case 'esc32a':
        backgroundColor = '#b2a1f6'
        break
      case 'G1':
        backgroundColor = '#9a916d'
        break
      case 'G11':
        backgroundColor = '#406a0f'
        break
      case 'G32':
        backgroundColor = '#004b31'
        break
      case 'K2000':
        backgroundColor = '#18729f'
        break
      case '9x9_h39_022':
        backgroundColor = '#9997c3'
        break
      case '9x9_h17_055':
        backgroundColor = '#9cf0ec'
        break
      case '16x16_easy_233':
        backgroundColor = '#ce843c'
        break
      case '16x16_hard_194':
        backgroundColor = '#2d326e'
        break
      case 'P-n16-k8':
        backgroundColor = '#76ed60'
        break
      case 'P-n19-k2':
        backgroundColor = '#d7f7e8'
        break
      case 'tai256c':
        backgroundColor = '#bb15d7'
        break
      case 'eil51':
        backgroundColor = '#87a7d8'
        break
      case 'A-n32-k5':
        backgroundColor = '#94053a'
        break
      case 'chr12a':
        backgroundColor = '#e5c8a6'
        break
      case 'chr18a':
        backgroundColor = '#e2f2c0'
        break
    }

    return backgroundColor
  }

  return Object.keys(problemInstances).map((className) => (
    <div
      key={`${key}_${className}`}
      style={{ margin: '5px 0 5px 0', display: 'flex' }}
    >
      <span style={{ marginRight: 10 }}>{className}:</span>
      {problemInstances[className]?.map((instance) => (
        <div
          key={`${key}_${className}_${instance}`}
          style={{ marginRight: 10 }}
        >
          <LinkButton
            label={instance}
            link={`/classes/${className}/instances/${instance}`}
            backgroundColor={getBackgroundColor(instance)}
          />
        </div>
      ))}
    </div>
  ))
}

export { renderClientsCell, renderProblemInstancesCell }
