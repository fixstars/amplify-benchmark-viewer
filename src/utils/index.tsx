export * from './DataGridCell'
export * from './statistics'
