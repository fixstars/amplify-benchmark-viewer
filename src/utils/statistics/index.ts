function quartile(data: readonly number[], q: number) {
  const sortedData = data.slice().sort((a, b) => a - b)
  const pos = Math.floor((sortedData.length - 1) * q)
  return sortedData[pos]
}
function quartile0(data: readonly number[]) {
  return quartile(data, 0)
}
function quartile25(data: readonly number[]) {
  return quartile(data, 0.25)
}

function quartile50(data: readonly number[]) {
  return quartile(data, 0.5)
}

function quartile75(data: readonly number[]) {
  return quartile(data, 0.75)
}

function quartile100(data: readonly number[]) {
  return quartile(data, 1)
}

export { quartile0, quartile25, quartile50, quartile75, quartile100 }
