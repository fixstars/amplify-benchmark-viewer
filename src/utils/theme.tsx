import { createTheme } from '@mui/material/styles'

export const theme = createTheme({
  palette: {
    primary: {
      main: '#111',
    },
  },
  typography: {
    h2: {
      fontSize: '1.5rem',
    },
  },
  components: {
    MuiTooltip: {
      styleOverrides: {
        tooltip: {
          backgroundColor: '#111',
          color: '#f4f1f4',
          font: '10px',
        },
      },
    },
  },
})
