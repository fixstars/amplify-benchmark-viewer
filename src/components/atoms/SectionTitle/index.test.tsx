import { render } from '@testing-library/react'
import { mockGrid } from 'utils/test'

import { SectionTitle } from '.'

describe('<SectionTitle />', () => {
  it('Rendered well', async () => {
    const { container } = render(<SectionTitle title="section title" />)

    const grid = mockGrid.mock.calls[0][0]
    expect(grid.item).toBe(true)
    expect(grid.xs).toBe(12)
    expect(grid.children.length).toBe(2)

    const title = grid.children[0]
    expect(title.type.render.displayName).toBe('Styled(Typography)')
    expect(title.props.variant).toBe('h2')
    expect(title.props.children).toBe('section title')

    const rightComponent = grid.children[1]
    expect(rightComponent).toBe(undefined)

    expect(container).toMatchSnapshot()
  })

  it('Right component is rendered well', async () => {
    render(
      <SectionTitle
        title="section title"
        rightComponent={<div>right component</div>}
      />,
    )

    const grid = mockGrid.mock.calls[0][0]
    expect(grid.children.length).toBe(2)

    const title = grid.children[0]
    expect(title.props.children).toBe('section title')

    const rightComponent = grid.children[1]
    expect(rightComponent.type).toBe('div')
    expect(rightComponent.props.children).toBe('right component')
  })
})
