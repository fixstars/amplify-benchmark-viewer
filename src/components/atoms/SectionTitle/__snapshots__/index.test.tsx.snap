// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTitle /> Rendered well 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-15h1z56-MuiGrid-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-140x6pm-MuiTypography-root"
    >
      section title
    </h2>
  </div>
</div>
`;
